File: SKIBin_old/app/src/main/java/com/h2Invent/skibin/ViewPageAdapter.java
package com.h2Invent.skibin;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

public class ViewPageAdapter extends FragmentStatePagerAdapter {
    private static int TAB_COUNT = 2;

    public ViewPageAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int position) {

        switch (position) {
            case 0:
                return FragmentConnection.newInstance();
            case 1:
                return FragmentSettings.newInstance();

        }
        return null;
    }

    @Override
    public int getCount() {
        return TAB_COUNT;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch (position) {
            case 0:
                return FragmentConnection.TITLE;

            case 1:
                return FragmentSettings.TITLE;

        }
        return super.getPageTitle(position);
    }
}

File: app/src/main/java/com/h2Invent/skibin/ViewPageAdapter.kt
package com.h2Invent.skibin

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

class ViewPageAdapter(fragmentManager: FragmentManager) :
    FragmentStatePagerAdapter(fragmentManager, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    val connectionFragment = FragmentConnection.newInstance()
    private val settingsFragment = FragmentSettings.newInstance()

    override fun getItem(position: Int): Fragment = when (position) {
        0 -> connectionFragment
        else -> settingsFragment
    }

    override fun getCount(): Int = 2

    override fun getPageTitle(position: Int): CharSequence = when (position) {
        0 -> FragmentConnection.TITLE
        else -> FragmentSettings.TITLE
    }
}
